dev/PushNotifications/PushNotificationTelemetry.h

```diff
@@ -116,21 +118,61 @@ class PushNotificationTelemetry : public wil::TraceLoggingProvider
         return m_eventCount;
     }
 
-    wchar_t m_appUserModelId[APPLICATION_USER_MODEL_ID_MAX_LENGTH] = {};
+    inline bool IsPackagedApp()
+    {
+        static bool isPackagedApp = AppModel::Identity::IsPackagedProcess();
+
+        return isPackagedApp;
+    }
+
+    inline const wchar_t* GetAppId()
+    {
+        static std::wstring appId = IsPackagedApp() ? InitAppIdPackaged() : InitAppIdUnpackaged();
+
+        return appId.c_str();
```

Comment:
Isn't this a bug? c_str() is scoped to appId lifetime.

Reply:
appId is a local static